### PR TITLE
Basic prevention of self-locking

### LINKED ISSUE #312: prevent users from locking areas where they have added all the candidates themselves
..or maybe make it impossible to do this if you have the usual 'lock area' permission, but have a special 'lock own areas' permission which only super admins have - just in case?

### PR DESCRIPTION
**Pull request by symroe**
Of course there are other ways to prevent this that are out of scope of
this PR.

For example, this doesn't cover the case where a user with locking
permission has added all the candidates themselves but hasn't suggested
locking. This case is harder to detect.

For the time being, this should catch the most obvious cases of
self-locking.

Closes #312

### CODE REVIEW

File: ynr/apps/elections/uk/templates/candidates/_candidates_for_post.html [hunk 1]
```diff
@@ -61,17 +61,31 @@
...
-
   {% if user_can_lock %}
     <form method="post" action="{% url 'constituency-lock' election=election %}">
       {% csrf_token %}
       {% for field in lock_form %}{{ field }}{% endfor %}
-      <input type="submit" class="button small" value="{% if candidates_locked %}{% trans "Unlock candidate list" %}{% else %}{% trans "Lock candidate list" %}{% endif %}">
+      {% if current_user_suggested_lock %}
+        <p>
+          Locking disabled because you suggested locking this post.
+          Please ask someone else to review it as a double check.
```
Comment (chris48s) [inline ynr/apps/elections/uk/templates/candidates/_candidates_for_post.html]: Happy with the logic, but a question about the messaging here:
Given there is no "request review" feature or messaging system or anything it seems like

> Please ask someone else to review it as a double check

is a CTA with no obvious path to fulfil it. i.e: As someone who has just suggested a lock here it sounds like I'm supposed to do something, but there's no obvious way to do it. Given users with appropriate permissions can see posts that need locking, would it be better to say something more like:

> Thanks for your contribution. An administrator will review the candidates for this area against the nomination paper

..or just not show anything except the absence of a 'suggest lock' button?

File: ynr/apps/elections/uk/tests/test_post_view.py [hunk 2]
```diff
@@ -80,3 +81,36 @@ def test_create_suggested_post_lock(self):
...
+        )
+
+        response = self.app.get(
+            '/election/2015/post/14419/edinburgh-east',
+            user=self.user,
+        )
+        self.assertContains(
+            response,
+            "Locking disabled because you suggested locking this post"
+        )
```
Comment (chris48s) [inline ynr/apps/elections/uk/tests/test_post_view.py]: Is it also worth bashing in a quick test for the inverse condition here - i.e: I **can** lock a post if **someone else** suggested the lock?

Comment (symroe) [reply ynr/apps/elections/uk/tests/test_post_view.py]: 👍

File: ynr/apps/elections/uk/templates/candidates/_candidates_for_post.html (hunk 1 above)
Comment (symroe) [reply ynr/apps/elections/uk/templates/candidates/_candidates_for_post.html]: Fair enough! I thought about linking to the Slackin but actually I think "Someone else will double check it soon." will do for now. Fixup pushed

**Pull request merged**